libs/core/langchain_core/utils/function_calling.py

```diff
@@ -187,6 +187,67 @@ def convert_python_function_to_openai_function(
     )
 
 
+def _convert_typed_dict_to_openai_function(typed_dict: Type) -> FunctionDescription:
+    model = cast(Type[BaseModel], _convert_typed_dict_to_pydantic(typed_dict))
+    return convert_pydantic_to_openai_function(model)
+
+
+_MAX_TYPED_DICT_RECURSION = 25
+
+
+def _convert_typed_dict_to_pydantic(type_: Type, *, depth: int = 0) -> Type:
```

Comment:
Is there anyway to leverage pydantic here?

Main issue seems to be doc-strings?

```python
from typing_extensions import TypedDict, Annotated

from pydantic import RootModel

class Meow(TypedDict):
"""Meow"""
c: str

class Foo(TypedDict):
a: Annotated[int, 'hello']
b: float
meow: Meow

Q = RootModel[Foo]

Q.schema()
```

Do you think we'll accept dataclasses or nested dataclasses / typed-dicts?

Reply:
from dataclasses import dataclass, field

@dataclass
class Foo:
a: int = field(metadata={'description': 'hello'})

    def add_one_to_a(self):
        return self.a + 1

RootModel[Foo].schema()

foo = RootModel[Foo].parse_obj({'a': 5}).root

foo.add_one_to_a()

Reply:
does pydantic v2 have RootModel?